spec/requests/api/orders/update_spec.rb

```diff
@@ -0,0 +1,44 @@
+RSpec.describe 'Update an order with a PUT /api/orders/:id', type: :request do
+
+  
+  let(:user) { create(:user) }
+  let(:product_1) { create(:product, name: 'Tshirt') }
+  let(:product_2) { create(:product, name: 'Football') }
+  let(:product_3) { create(:product, name: 'Scarf') }
+  let(:order) { create(:order) }
+  let!(:order_item_1) { create(:order_item, order:, product: product_2) }
+  let!(:order_item_2) { create(:order_item, order:, product: product_3) }
+
+  subject { response }
+  
+  describe 'succesful order' do
+  before do
+    put "/api/orders/#{order.id}", params: { product_id: product_2.id }
+
+    @order = Order.last
+  end
+
+  it { is_expected.to have_http_status :ok }
+
+  it 'response with an order confirmation' do
+    response_body = JSON.parse(response.body)
+    expect(response_body['message']).to eq 'Football added'
+  end
+
+  it 'is expected to contain 3 items' do
+    expect(@order.items.count).to eq 3
+  end
+
```

Comment:
Consider adding a test to make sure that the order includes the items you expect it to include. If I read the above correctly, Tshirt is never added.

Reply:
<img width="842" alt="Screenshot 2022-05-04 at 19 33 15" src="https://user-images.githubusercontent.com/92160782/166745850-bae28992-b0ef-4924-af5e-ce9fb611c0cf.png">

Reply:
Updated, did scarf by accident!